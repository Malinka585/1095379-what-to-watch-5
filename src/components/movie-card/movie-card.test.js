import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";

import MovieCard from './movie-card';

const film = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `img/the-grand-budapest-hotel-bg.jpg`,
  genre: `Comedy`,
  released: 2014,
};

const noop = () => {};

it(`MovieCard is rendered correctly`, () => {

  const tree = renderer.create(
      <BrowserRouter>
        <MovieCard
          onFilmCardOver={noop}
          onFilmCardLeave={noop}
          film={film}
        >
          <video/>
        </MovieCard>
      </BrowserRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
