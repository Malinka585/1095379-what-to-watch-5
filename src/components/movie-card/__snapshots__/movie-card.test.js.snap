// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard is rendered correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  id={1}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <video />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      The Grand Budapest Hotel
    </a>
  </h3>
</article>
`;
